slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -329,6 +332,109 @@ void SLAMServiceImpl::SetUpMapBuilder() {
 }
 
 std::string SLAMServiceImpl::GetLatestJpegMapString(bool add_pose_marker) {
+    cartographer::io::PaintSubmapSlicesResult painted_slices =
+        GetLatestPaintedMapSlices();
+    if (add_pose_marker) {
+        PaintMarker(painted_slices);
+    }
+
+    auto image = io::Image(std::move(painted_slices.surface));
+    return image.WriteJpegToString(50);
+}
+
+bool SLAMServiceImpl::GetLatestPointCloudMapString(std::string &pointcloud) {
+    cartographer::io::PaintSubmapSlicesResult painted_slices =
+        GetLatestPaintedMapSlices();
+    auto painted_surface = painted_slices.surface.get();
+    int width = cairo_image_surface_get_width(painted_surface);
+    int height = cairo_image_surface_get_height(painted_surface);
+
+    // total number of bytes, 32 bits per pixel
+    size_t size_data = width * height * 4;
+    auto data = cairo_image_surface_get_data(painted_surface);
+
+    std::vector<unsigned char> data_vect(data, data + size_data);
+
+    std::stringbuf data_buffer;
+
+    int num_points = 0;
+
+    // Reduce resolution based off number of pixels. Output is pixels to skip
+    int scaleFactor = 4;
+    int scale = size_data / 100000.f / 32.f / scaleFactor;
+
+    // Loop to filter unwanted data and reduce resolution
+    for (int i = 0; i < size_data; i = i + scale * 4) {
+        int rgb = 0;
+        rgb = rgb | ((int)data_vect[i + 0] << 16);
+        rgb = rgb | ((int)data_vect[i + 1] << 8);
+        rgb = rgb | ((int)data_vect[i + 2] << 0);
+
+        // skip pixels that are not in our map(black/past walls)
+        //  this value represents [102,102,102]
+        if (rgb == 6710886) continue;
+
+        // Determine probability based off color pixel
+        int prob = ViamColorToProbability((int)data_vect[i + 2]);
+
+        if (prob == 0) continue;
+
+        num_points++;
+
+        int pixel_index = i / 4;
+        int pixel_x = pixel_index % width;
+        int pixel_y = pixel_index / width;
+
+        // kPixelSize is .01, unsure of units but assumed converts to meters
+        // based on PaintSubmapSlices() and DrawPoseOnSurface()
+        float x_pos = (pixel_x - painted_slices.origin.x()) * kPixelSize;
+        // Y is inverted to match output from getPosition()
+        float y_pos = -(pixel_y - painted_slices.origin.y()) * kPixelSize;
```

Comment:
It would be nice to solve this problem here, if this just allows the position to show up correct in relation to the map but it is still reversed, i might advise making the change needed in position

Reply:
not sure I know what the "map showing up in reverse" issue is, was it related to the jpeg output or the pcd output?

Reply:
The jpeg output, e.g. the office map showing up with flipped orientation. We'll need to ensure that by the end of this project, the orientation is correct in the jpeg map produced by rdk. (Not necessarily this ticket, if you're not able to test this yet.)
<img width="445" alt="map fixed" src="https://user-images.githubusercontent.com/13758212/215533560-bdc46934-1544-4f58-a2f8-023701fefa83.png">

Reply:
good question john. The whole PCD is reversed to my eyes: 
![image](https://user-images.githubusercontent.com/92594844/215871861-a8b572cb-c7ac-4571-a577-2e6da4bfba95.png)

Reply:
this PR does not fix this behavior